### DISCUSSION TITLE
How to select current const statement in TypeScript VIM?

### DISCUSSION THREAD

**Discussion by jmagaram** [zed-industries/zed#22344] [Feature Requests]
I'm struggling with how to select chunks of code based on syntax analysis. If I put the cursor on `: kind === "virtualKey"` and want to select the entire STATEMENT that defines `adapter`, how do I do that in VIM? I've tried `via`, `vaf`, etc. and nothing works. The best I've got it repeatedly pressing `v[x[x[x`.

```typescript
  const [os, setOs] = useState<OperatingSystem>("mac");
  const [kind, setKind] = useState<KeyViewType>("scanCode");
  const adapter =
    kind === "producedCharacter"
      ? producedCharacterAdapter
      : kind === "virtualKey"
        ? virtualKeyAdapter
        : scanCodeAdapter;
```

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
